enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/state/snapshot/CoreStateDownloaderService.java

```diff
@@ -52,20 +56,38 @@ public CoreStateDownloaderService( JobScheduler jobScheduler, CoreStateDownloade
 
     public synchronized void scheduleDownload( LeaderLocator leaderLocator )
     {
+        if ( stopped )
+        {
+            return;
+        }
+
         if ( currentJob == null || currentJob.hasCompleted() )
         {
             currentJob = new PersistentSnapshotDownloader( leaderLocator, applicationProcess, downloader, log,
                     downloaderPauseStrategy );
-            jobScheduler.schedule( downloadSnapshot, currentJob );
+            jobHandle = jobScheduler.schedule( downloadSnapshot, currentJob );
         }
     }
 
     @Override
     public synchronized void stop() throws Throwable
     {
-        if (currentJob != null)
+        stopped = true;
```

Comment:
Do we need to implement start which sets stopped=false? Or will this entire object be recreated?

Reply:
I think that is flexible, contract-wise. We do not expect this particular object to be restartable. The service has he same lifespan as the entire database process.